### PR TITLE
Enhance error msg of imperative code

### PR DESCRIPTION
**Pull request by zhiqiu**
ATT， refine the error message of c++ imperative code.

### CODE REVIEW

File: paddle/fluid/imperative/layer.cc [hunk 1]
```diff
@@ -41,7 +41,8 @@ void ThreadSafeNameSet::Insert(const std::string& name) {
 void ThreadSafeNameSet::Remove(const std::string& name) {
   std::lock_guard<std::mutex> guard(mtx_);
   auto iter = set_.find(name);
-  PADDLE_ENFORCE_EQ(iter != set_.end(), true, "%s does not exist", name);
+  PADDLE_ENFORCE_EQ(iter != set_.end(), true,
+                    platform::errors::NotFound("Name %s does not exist", name));
```
Comment (chenwhql) [inline paddle/fluid/imperative/layer.cc]: who's name? Variable name?

File: paddle/fluid/imperative/tests/test_prepare_op.cc [hunk 2]
```diff
@@ -68,8 +68,9 @@ static framework::VariableNameMap CreateVarNameMap(
     if (it == varbase_map.end()) {
       PADDLE_ENFORCE_EQ(
           var.dispensable(), true,
-          "Var: %s not dispensable and there are no such var in inputs",
-          var.name());
+          platform::errors::NotFound(
+              "Var: %s is not dispensable and there are no such var in inputs",
```
Comment (chenwhql) [inline paddle/fluid/imperative/tests/test_prepare_op.cc]: `var` -> `Variable(VarBase)`?

File: paddle/fluid/pybind/imperative.cc [hunk 3]
```diff
@@ -99,9 +99,8 @@ static void InitTensorForVarBase(imperative::VarBase *self,
 
 static void InitVarBaseFromNumpyWithKwargs(imperative::VarBase *self,
                                            const py::kwargs &kwargs) {
-  PADDLE_ENFORCE_EQ(
-      kwargs.contains("value"), true,
-      platform::errors::InvalidArgument("Missing argument: value"));
+  PADDLE_ENFORCE_EQ(kwargs.contains("value"), true,
+                    platform::errors::NotFound("Missing argument: value"));
```
Comment (chenwhql) [inline paddle/fluid/pybind/imperative.cc]: what missing argument?

File: paddle/fluid/imperative/tests/test_prepare_op.cc (hunk 2 above)
Comment (zhiqiu) [reply paddle/fluid/imperative/tests/test_prepare_op.cc]: Done, thx.

File: paddle/fluid/pybind/imperative.cc (hunk 3 above)
Comment (zhiqiu) [reply paddle/fluid/pybind/imperative.cc]: Done, thx.

File: paddle/fluid/imperative/layer.cc (hunk 1 above)
Comment (zhiqiu) [reply paddle/fluid/imperative/layer.cc]: Done, thx.

**Review by sneaxiy — APPROVED** [review]

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**